addons/account/models/partner.py

```diff
@@ -314,6 +314,7 @@ class AccountFiscalPositionAccount(models.Model):
 class ResPartner(models.Model):
     _name = 'res.partner'
     _inherit = 'res.partner'
+    _check_company_domain = models.check_company_domain_parent_of
```

Comment:
Why in `account` and not in `base`, the security rule is set in `odoo/addons/base/security/base_security.xml`